### PR TITLE
UICIRC-365: Fix data refreshing in the circulation rules editor

### PR DESCRIPTION
**Pull request by YevheniiMaltsev**
## Purposes

Fix data refreshing in the circulation rules editor in the scope of the [ticket](https://issues.folio.org/browse/UICIRC-365).

### CODE REVIEW

File: src/settings/CirculationRules.js [hunk 1]
```diff
@@ -168,14 +168,89 @@ class CirculationRules extends React.Component {
...
     this.onSubmit = this.onSubmit.bind(this);
-    this.state = {};
+
+    this.state = {
+      isLoading: true,
+    };
+  }
+
+  componentDidMount() {
+    const resourcesFields = [
```
[the remark this answers is not in the document]
Comment (YevheniiMaltsev) [reply src/settings/CirculationRules.js]: @VictorSoroka1 thanks for your comment. Done!

[the remark this answers is not in the document]
Comment (VictorSoroka1) [reply src/settings/CirculationRules.js]: Thank you.

**Comment by mkuklis** [comment]
Oh I see. I think we run into this issue before and `resourceShouldRefresh` should do the right thing here and refresh the data. If this doesn't happen then we should try to understand what is going on.

**Comment by YevheniiMaltsev** [comment]
@zburke @mkuklis  thanks for your comments! 👍 

The issue (UICIRC-365) is reproduced in the following the next steps (For example):

1. Open the circulation rules editor (Settings -> Circulation -> Circulation rules)
2. Go to  Settings -> Inventory-> Material types to change/add a material type
3. Open the circulation rules editor again (Settings -> Circulation -> Circulation rules)

Without the provided changes, the editor handled the step 3 in a next way:

- The application renders the editor using the outdated data (they were loaded on step 1) that are stored in the mutators.
- RulesEditor component (src -> settings -> lib-> RulesEditor -> RulesEditor.js) init the rules editor hints in the constructor.
- If the fresh data received the ui is rerendered, but the rules hint data is not refreshed because it was inited on the previous step in the constructor.

So, in general, this issue can be fixed by putting new data to the editor (Move the editor setup from the constructor, etc.). But if there is a delay on the server or a slow connection, it will take some time to reload the data, and a user will see an incorrect state or dropdowns reloading (hints), etc. It was agreed to reset data and render the circulation rules editor only in case of refreshed data, and we are going to propose adding a loading indicator to display it instead of the empty pane. A similar solution is used in the other places of the applications, and I wasn't able to find a way to reset data in the API description (https://github.com/folio-org/stripes-connect/blob/master/doc/api.md). But if there are better solutions to handle the case like this, I will be glad to implement them :)

**Comment by zburke** [comment]
stripes-connect implements `reset = () => dispatch(this.actions.reset()),` on a resource only when it is configured with `accumulate: true`. Is the right approach here to provide `reset` to all resources, and to call that in `componentDidMount`? 

i.e. we don't really need manual fetching here; we just need to make sure the data we are already retrieving is what the UI displays.

**Comment by mkuklis** [comment]
@YevheniiMaltsev thank you for your detailed description! You are right that this is currently a limitation in `stripes-connect` (and it's a frustrating one).

@zburke and I spent a bit more time with this and we patched stripes-connect a bit: https://github.com/folio-org/stripes-connect/pull/112

which should make sure the resources with `resourceShouldRefresh` are reset when the connected component is unmounted. 

We are hoping that this change together with the little tweak to `CirculationRules` render (https://github.com/folio-org/ui-circulation/pull/477) will resolve the issue described here.

**Comment by zburke** [comment]
Replaced by #477. 

Seeing all the manual fetches in this PR was the straw that broke the camel's back for me WRT how stripes-connect handles `resourceShouldRefresh` (er, how it _failed_ to handle it). So even though I'm closing this, I'm really glad to do so because it caused us to fix the underlying problem (see folio-org/stripes-connect/pull/112 for details).

**Pull request closed without merging**